mapie/conformity_scores/utils.py

```diff
@@ -1,26 +1,163 @@
+        )
+
+
+def _check_depreciated(
+    method: str
+) -> None:
+    """
+    Check if the chosen method is outdated.
+
+    Raises
+    ------
+    Warning
+        If method is ``"score"`` (not ``"lac"``) or
+        if method is ``"cumulated_score"`` (not ``"aps"``).
+    """
+    if method == "score":
+        warnings.warn(
+            "WARNING: Deprecated method. "
+            + "The method \"score\" is outdated. "
+            + "Prefer to use \"lac\" instead to keep "
+            + "the same behavior in the next release.",
+            DeprecationWarning
+        )
+    if method == "cumulated_score":
+        warnings.warn(
+            "WARNING: Deprecated method. "
+            + "The method \"cumulated_score\" is outdated. "
+            + "Prefer to use \"aps\" instead to keep "
+            + "the same behavior in the next release.",
+            DeprecationWarning
+        )
+
+
+def check_target(
+    conformity_score: BaseClassificationScore,
+    y: ArrayLike
+) -> None:
+    """
+    Check that if the type of target is binary,
+    (then the method have to be ``"lac"``), or multi-class.
 
     Parameters
     ----------
-    y_pred_proba: NDArray of shape (n_samples, n_classes)
-        Model prediction.
+    conformity_score: BaseClassificationScore
+        Conformity score function.
 
-    y: NDArray of shape (n_samples)
-        Labels.
+    y: NDArray of shape (n_samples,)
+        Training labels.
 
-    Returns
-    -------
-    NDArray of shape (n_samples, 1)
-        Position of the true label in the prediction.
+    Raises
+    ------
+    ValueError
+        If type of target is binary and method is not ``"lac"``
+        or ``"score"`` or if type of target is not multi-class.
     """
-    index = np.argsort(np.fliplr(np.argsort(y_pred_proba, axis=1)))
-    position = np.take_along_axis(index, y.reshape(-1, 1), axis=1)
+    check_classification_targets(y)
+    if type_of_target(y) == "binary" and not isinstance(conformity_score, LAC):
+        raise ValueError(
+            "Invalid method for binary target. "
+            "Your target is not of type multiclass and "
+            "allowed values for binary type are "
+            f"{['score', 'lac']}."
+        )
+
 
-    return position
+def check_classification_conformity_score(
+    conformity_score: Optional[BaseClassificationScore] = None,
+    method: Optional[str] = None,
+) -> BaseClassificationScore:
+    """
+    Check parameter ``conformity_score`` for classification task.
+
+    Raises
+    ------
+    ValueError
+        If parameters are not valid.
+
+    Examples
+    --------
+    >>> from mapie.conformity_scores.utils import (
+    ...     check_classification_conformity_score
+    ... )
+    >>> try:
+    ...     check_classification_conformity_score(1)
+    ... except Exception as exception:
+    ...     print(exception)
+    ...
+    Invalid conformity_score argument.
+    Must be None or a ConformityScore instance.
+    """
```

Comment:
Maybe cleaner if you create a dictionnary to map method to conformity scores

Reply:
Moreover, as allowed methods are also defined in MapieClassifier, the values for `allowed_methods` should be an argument of this function

Reply:
Done 👍 everything (like `allowed_methods`) is managed directly by the check function

Reply:
I think that the allowed methods should be handled in MapieClassifier

Reply:
No, I think it's better to keep a black box perspective. And it remains within the logic of outsourcing the scores with MapieClassifier.